### ISSUE TITLE
serial-vault snap should be easier to use

### ISSUE THREAD

**Issue by rmescandon** [ubuntu-core/identity-vault#74]
So, in general I'm not too happy with the UX:

Users have to provide settings.yaml to config command and then restart the service. That's not bad, but some of the settings are not going to be changed (title, logo, docRoot, driver) or maybe yes, but not frequently. And others maybe require a quicker switch; for instance, think in starting server in admin mode and later restart in system-user one. That requires annoying steps of reconfigure and restart service. I wonder if we could use the snap config mechanism (https://github.com/snapcore/snapd/wiki/hooks) to modify that (or any other) value, so that there only would be needed set a specific value for the mode and restart.

*cat settings.yaml | sudo serial-vault.config && sudo systemclt restart ...*
or
*snap set serial-vault mode=system-user && sudo systemctl restart ...*

basically it is the same, but second sounds better.

Other thing. This way, if I access behind https using a domain, that would work. If I access directly skipping csrf... well, it's a pain setting an env var (actually for root, as that's the user starting the service). Using snap config can make this easier to enable/disable.

And finally, we don't have a kind of automatic way of generating csrfAuthKey or apiKeys. Maybe commands could generate them and update settings.yaml at the same time the value is printed out:

$ serial-vault.generate-api-key
U2VyaWFsIFZhdWx0Cg

So, in general, having a way to modify directly settings file sounds good, by maybe we should provide a wider range of options to users

**Comment by slimjim777** [comment]
I agree on the UX. The clunky way of defining config was the only way that was available at the time the snap was written. Updating that to use the latest features like 'snap set' and the configure hook would be better.

An enable/disable of the config for csrf would make things simpler.

The automated generation of the csrfAuthKey and API key could be problematic in a data centre where there and multiple units serving the same service e.g. admin or signing service. Since the config will need to be the same between the units, getting the service to default to a setting may not be the best approach.

I would like to make the configuration simpler, but I think that is difficult in some deployment environments, especially since we have multiple services that need to share config. This could certainly work for the snap bundle that installs the 3 services, postgresql and apache in a single snap.

**Comment by slimjim777** [comment]
In practice, our recommended approach is to use juju with one of the charm bundles. That greatly simplifies deployment and has the type of config setting that you are suggesting, including the auto-restart of a service on config change.

In some ways, the snap install is a lower-level installation and maybe recommended more for those that want a highly customised deployment. The juju bundle greatly simplifies deployment and configuration setting.

**Comment by slimjim777** [comment]
This is probably no longer relevant as we will have a factory version of the serial-vault that syncs with the cloud version. The factory version will be delivered as a snap. The cloud version is available as a charm and charm bundle - neither of which will use the snap.

**Comment by slimjim777** [comment]
The factory-serial-vault snap is now done. The UI is simpler -  can we close this?
